staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances.go

```diff
@@ -310,17 +314,20 @@ func (az *Cloud) InstanceMetadataByProviderID(ctx context.Context, providerID st
 }
 
 func (az *Cloud) isCurrentInstance(name types.NodeName, metadataVMName string) (bool, error) {
+	var err error
 	nodeName := mapNodeNameToVMName(name)
 
+	// VMSS vmName is not same with hostname, use hostname instead.
 	if az.VMType == vmTypeVMSS {
-		// VMSS vmName is not same with hostname, construct the node name "{computer-name-prefix}{base-36-instance-id}".
-		// Refer https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-instance-ids#scale-set-vm-computer-name.
-		if ssName, instanceID, err := extractVmssVMName(metadataVMName); err == nil {
-			instance, err := strconv.ParseInt(instanceID, 10, 64)
-			if err != nil {
-				return false, fmt.Errorf("failed to parse VMSS instanceID %q: %v", instanceID, err)
-			}
-			metadataVMName = fmt.Sprintf("%s%06s", ssName, strconv.FormatInt(instance, 36))
+		metadataVMName, err = os.Hostname()
+		if err != nil {
+			return false, err
+		}
+
+		// Use name from env variable "NODE_NAME" if it is set.
```

Comment:
in the PR, said `HOST_NAME`, while it's `NODE_NAME` here? also, is there code change needed on aks & aks-engine, etc.?

Reply:
/hold 
hold a while for clarification, thanks.

Reply:
fixed, should be "NODE_NAME". No changes are required from other components.